lib/screens/authenticate/forgotpass.dart

```diff
@@ -0,0 +1,85 @@
+import 'package:flutter/material.dart';
+import 'package:shareacab/services/auth.dart';
+import 'package:shareacab/shared/constants.dart';
+import 'package:shareacab/shared/loading.dart';
+
+class ForgotPass extends StatefulWidget {
+  @override
+  _ForgotPassState createState() => _ForgotPassState();
+}
+
+class _ForgotPassState extends State<ForgotPass> {
+  final AuthService _auth = AuthService();
+  final _formKey = GlobalKey<FormState>();
+  bool loading = false;
+
+  String email = '';
+  String message = '';
+
+  @override
+  Widget build(BuildContext context) {
+    return loading
+        ? Loading()
+        : Scaffold(
+            backgroundColor: Colors.brown[100],
+            appBar: AppBar(
+              backgroundColor: Colors.brown[400],
+              elevation: 0.0,
+              title: Text('Forgot Password'),
+            ),
+            body: Container(
+              padding: EdgeInsets.symmetric(vertical: 20.0, horizontal: 50.0),
+              child: Form(
+                key: _formKey,
+                child: SingleChildScrollView(
+                  child: Column(
+                    children: <Widget>[
+                      SizedBox(height: 20.0),
+                      TextFormField(
+                        decoration:
+                            textInputDecoration.copyWith(hintText: 'Email'),
+                        validator: (val) =>
+                            val.isEmpty ? 'Enter a valid Email' : null,
+                        onChanged: (val) {
+                          setState(() => email = val);
+                        },
+                      ),
+                      SizedBox(height: 20.0),
+                      RaisedButton(
+                        color: Colors.pink[400],
+                        child: Text(
+                          'Send Password Reset Link',
+                          style: TextStyle(color: Colors.white),
+                        ),
+                        onPressed: () async {
+                          if (_formKey.currentState.validate()) {
+                            setState(() => loading = true);
+                            await _auth.resetPassword(email);
```

Comment:
resetPassword returns a Future, you also have to check if it failed. See [this](https://api.dart.dev/stable/2.8.2/dart-async/Future-class.html), basically you need to do a catchError either here or in the resetPassword function. And if there is an error, you need to display it too

Reply:
fixed in the latest PR